src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/RestDataSourceItemFixture.cs

```diff
@@ -404,5 +408,95 @@ public void UseCsv_Should_Add_TwoDataSources()
             Assert.Equal(DataSourceIds.CSV, document.DataSources[0].Id);
             Assert.Equal(DataSourceProvider.CSV, document.DataSources[0].Provider);
         }
+
+        [Fact]
+        public void RDashDocument_HasCorrectDataSourceItem_WhenLoadFromFile()
+        {
+            // Arrange
+            var filePath = Path.Combine(Environment.CurrentDirectory, "Dashboards", "TestRest.rdash");
+
+            // Act
+            var document = RdashDocument.Load(filePath);
+            var dataSource = document.DataSources.LastOrDefault();
+            var dataSourceItem = document.Visualizations.LastOrDefault().DataDefinition.DataSourceItem;
+
+            // Assert
+            Assert.Equal(DataSourceProvider.REST, dataSource.Provider);
+            Assert.False(dataSourceItem.Properties.GetValue<bool>("ServerAggregation"));
+        }
+
+        [Fact]
+        public void ToJsonString_CreatesFormattedJson_ForWebServiceDataSource()
+        {
+            // Arrange
+            var expectedJson = @"
+            {
+              ""_type"": ""DataSourceItemType"",
+              ""Id"": ""RestItem"",
+              ""Title"": ""Rest DS Item"",
+              ""DataSourceId"": ""__JSON"",
+              ""HasTabularData"": true,
+              ""HasAsset"": false,
+              ""Properties"": {},
+              ""Parameters"": {
+                ""config"": {
+                  ""iterationDepth"": 0,
+                  ""columnsConfig"": [
+                    {
+                      ""key"": ""_id"",
+                      ""type"": 0
+                    },
+                    {
+                      ""key"": ""name"",
+                      ""type"": 0
+                    }
+                  ]
+                }
+              },
+              ""ResourceItem"": {
+                ""_type"": ""DataSourceItemType"",
+                ""Id"": ""RestItem"",
+                ""Title"": ""DB Test"",
+                ""DataSourceId"": ""Rest"",
+                ""HasTabularData"": true,
+                ""HasAsset"": false,
+                ""Properties"": {},
+                ""Parameters"": {}
+              }
+            }";
+
+            var dataSource = new RestDataSource()
```

Comment:
this is an internal class and should not be used to create the REST data source item. You should be using the URL properties on the rest data source item itself. If you need a data source to pass in, pass in a base data source object.